plugins/usi-dock/fu-usi-dock-dmc-device.c

```diff
@@ -55,6 +58,28 @@ fu_usi_dock_dmc_device_parent_notify_cb(FuDevice *device, GParamSpec *pspec, gpo
 			return;
 		}
 
+		/* allow matching PCB version */
+		serialnum = fu_device_get_serial(device);
+		if (strlen(serialnum) <= USI_SN_NUMBER_LENTH) {
```

Comment:
you need to test that the serial number is long enough to do `serialnum[7]`, e.g. `if (strlen(serialnum) >= 8) {`